Cubical/Cohomology/EilenbergMacLane/Base.agda

```diff
@@ -99,6 +111,163 @@ is-set (isSemigroup (isMonoid (isGroup (isAbGroup (snd (coHomGr n G A)))))) = sq
+               (cong₂ _+ₖ_ (sym f1) (cong -ₖ_ (sym f1) ∙ -0ₖ n)
+                                  ∙ rUnitₖ n (0ₖ n))
+               refl
+    help zero _ _ = isOfHLevelPathP' 0 (is-set (snd G) _ _) _ _ .fst
+    help (suc zero) = J> (sym (rUnit _) ∙ sym (rUnit _))
+    help (suc (suc n)) = J> (sym (rUnit _) ∙ sym (rUnit _))
+
+  lCancelₕ∙ : (x : coHomRed n G A) → ((-ₕ∙ x) +ₕ∙ x) ≡ 0ₕ∙ n
+  lCancelₕ∙ x = commₕ∙ (-ₕ∙ x) x ∙ rCancelₕ∙ x
+
+  rUnitₕ∙ : (x : coHomRed n G A) → (x +ₕ∙ 0ₕ∙ n) ≡ x
+  rUnitₕ∙ = ST.elim (λ _ → isSetPathImplicit)
+     λ f → cong ∣_∣₂ (ΣPathP ((funExt λ x → rUnitₖ n (fst f x))
+                   , help n _ (sym (snd f))))
+    where
+    help : (n : ℕ) (f0 : EM G n) (f1 : 0ₖ n ≡ f0)
+      → PathP (λ i → rUnitₖ n f0 i ≡ 0ₖ n)
+               (cong (_+ₖ 0ₖ n) (sym f1) ∙ rUnitₖ n (0ₖ n))
+               (sym f1)
+    help zero _ _ = isOfHLevelPathP' 0 (is-set (snd G) _ _) _ _ .fst
+    help (suc zero) = J> sym (rUnit refl)
+    help (suc (suc n)) = J> sym (rUnit refl)
+
+  lUnitₕ∙ : (x : coHomRed n G A) → (0ₕ∙ n +ₕ∙ x) ≡ x
+  lUnitₕ∙ = ST.elim (λ _ → isSetPathImplicit)
+    λ f → cong ∣_∣₂ (ΣPathP ((funExt (λ x → lUnitₖ n (fst f x)))
+                  , help n _ (sym (snd f))))
+    where
+    help : (n : ℕ) (f0 : EM G n) (f1 : 0ₖ n ≡ f0)
+      → PathP (λ i → lUnitₖ n f0 i ≡ 0ₖ n)
+               (cong (0ₖ n +ₖ_) (sym f1) ∙ rUnitₖ n (0ₖ n))
+               (sym f1)
+    help zero _ _ = isOfHLevelPathP' 0 (is-set (snd G) _ _) _ _ .fst
+    help (suc zero) = J> sym (rUnit refl)
+    help (suc (suc n)) = J> sym (rUnit refl)
+
+  assocₕ∙ : (x y z : coHomRed n G A) → x +ₕ∙ (y +ₕ∙ z) ≡ (x +ₕ∙ y) +ₕ∙ z
+  assocₕ∙ =
+    ST.elim3 (λ _ _ _ → isSetPathImplicit)
+     λ f g h → cong ∣_∣₂
+       (ΣPathP ((funExt (λ x → assocₖ n (fst f x) (fst g x) (fst h x)))
+      , help n _ (sym (snd f)) _ (sym (snd g)) _ (sym (snd h))))
+    where
+    help : (n : ℕ) (f0 : EM G n) (f1 : 0ₖ n ≡ f0)
+                    (g0 : EM G n) (g1 : 0ₖ n ≡ g0)
+                    (h0 : EM G n) (h1 : 0ₖ n ≡ h0)
+        → PathP (λ i → assocₖ n f0 g0 h0 i ≡ 0ₖ n)
+                 (cong₂ _+ₖ_ (sym f1)
+                  (cong₂ _+ₖ_ (sym g1) (sym h1) ∙ rUnitₖ n (0ₖ n))
+                    ∙ rUnitₖ n (0ₖ n))
+                 (cong₂ _+ₖ_ (cong₂ _+ₖ_ (sym f1) (sym g1)
+                   ∙ rUnitₖ n (0ₖ n)) (sym h1) ∙ rUnitₖ n (0ₖ n))
+    help zero _ _ _ _ _ _ = isOfHLevelPathP' 0 (is-set (snd G) _ _) _ _ .fst
+    help (suc zero) =
+      J> (J> (J> cong (_∙ refl)
+        (cong (cong₂ _+ₖ_ refl) (sym (rUnit refl))
+        ∙ (cong (λ z → cong₂ (+ₖ-syntax {G = G} 1) z (refl {x = 0ₖ {G = G} 1}))
+          (rUnit refl)))))
+    help (suc (suc n)) =
+      J> (J> (J> flipSquare ((sym (rUnit refl))
+        ◁ flipSquare (cong (_∙ refl)
+        (cong (cong₂ _+ₖ_ refl) (sym (rUnit refl))
+        ∙ (cong (λ z → cong₂ (+ₖ-syntax {G = G} (suc (suc n))) z
+                              (refl {x = 0ₖ {G = G} (suc (suc n))}))
+          (rUnit refl)))))))
```

Comment:
I think these names should be in a named module or private block, so it is clear they are not supposed to be exported.